### PR TITLE
Blob Storage download

### LINKED ISSUE #3478: Update download_manager.stream_tarred_gzipped_directory to work with Blob Storage
Update the logic in download_manager.stream_tarred_gzipped_directory to work with Blob Storage. Right now, this function only works with local paths (see https://github.com/codalab/codalab-worksheets/blob/5f72f4d2d86857ccef2abc016a2b071c1f3dc54c/codalab/lib/download_manager.py#L170). We need to modify it to use the `OpenFile` context manager instead.

### PR DESCRIPTION
**Pull request by epicfaace**
Implements download_manager changes needed for blob storage. Fixes https://github.com/codalab/codalab-worksheets/issues/3478.

This PR is based on https://github.com/codalab/codalab-worksheets/pull/3496/

### CODE REVIEW

File: codalab/lib/beam/ratarmount.py [hunk 1]
```diff
@@ -753,6 +753,7 @@ def _createIndex(
                     # fmt:on
                 )
             except tarfile.ReadError:
+                print("RE")
```
Comment (adiprerepa) [inline codalab/lib/beam/ratarmount.py]: is this the correct way to handle this? Might be ambiguous to the reader
```suggestion
                print("Tar read error")
```

File: codalab/lib/upload_manager.py [hunk 2]
```diff
@@ -53,6 +53,7 @@ def upload_to_bundle_store(self, bundle: Bundle, source: Source, git: bool, unpa
         """Uploads the given source to the bundle store.
         Given arguments are the same as UploadManager.upload_to_bundle_store()"""
         try:
+            # bundle_path = self._bundle_store.get_bundle_location(bundle.uuid)
```
Comment (adiprerepa) [inline codalab/lib/upload_manager.py]: remove lol
```suggestion
```

File: codalab/worker/file_util.py [hunk 3]
```diff
@@ -268,7 +268,15 @@ def __enter__(self) -> IO[bytes]:
...
                     return GzipStream(fs) if self.gzipped else fs
         else:
-            # Stream a file from disk storage.
+            # Stream a directory or file from disk storage.
+            if os.path.isdir(self.path):
+                if not self.gzipped:
+                    raise IOError("Directories must be gzipped.")
+                return tar_gzip_directory(self.path)
+            if self.gzipped:
+                raise IOError(
```
Comment (adiprerepa) [inline codalab/worker/file_util.py:278]: does it make sense to add a comment to the function so future callers know this?

**Review by adiprerepa — COMMENTED** [review]
awesome sauce

File: codalab/lib/beam/ratarmount.py (hunk 1 above)
Comment (epicfaace) [inline codalab/lib/beam/ratarmount.py]:
```suggestion
```

Comment (epicfaace) [reply codalab/lib/beam/ratarmount.py]: I meant to remove this. This was the original code from the ratarmount library and I just added it in for debugging.

File: codalab/worker/file_util.py (hunk 3 above)
Comment (epicfaace) [reply codalab/worker/file_util.py:278]: done

**Review by adiprerepa — APPROVED** [review]

**Pull request merged**